hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/AggregateSlidingWindowPhysicalRule.java

```diff
@@ -236,6 +238,11 @@ private static Integer findWatermarkedField(
         if (watermarkedFields == null) {
             return null;
         }
-        return watermarkedFields.findFirst(logicalAggregate.getGroupSet());
+        ImmutableBitSet windowBoundIndexes = ImmutableBitSet.builder()
+                .addAll(windowStartIndexes)
+                .addAll(windowEndIndexes)
+                .build()
+                .intersect(logicalAggregate.getGroupSet());
```

Comment:
From what I see the builder also has ```intersect``` method, that will save us creating some litter. The ```intersect``` in ```ImmutableBitSet```:
```java
  public ImmutableBitSet intersect(ImmutableBitSet that) {
    final Builder builder = rebuild();
    builder.intersect(that);
    return builder.build();
  }
```
It just creates a builder. Unfortunately the ```intersect``` in builder doesn't returns it's instance.